neuralmonkey/decoders/decoder.py

```diff
@@ -398,15 +269,17 @@ def body(*args) -> LoopState:
                 # Run the RNN.
                 cell = self._get_rnn_cell()
                 if self._rnn_cell_str == 'GRU':
-                    cell_output, state = cell(rnn_input,
-                                              loop_state.prev_rnn_output)
+                    cell_output, state = cell(
+                        rnn_input, loop_state.feedables.prev_rnn_output)
```

Comment:
Toto tady mame uz dlouhou, ale i tak se zeptam. Skutecne tam ma byt jako druhy argument prev_rnn_output (misto prev_rnn_state). Pokud jo, tak u GRU dekoderu vubec prev_rnn_state nevyuzivame.

Nebo jsem neco prehleld?

Reply:
V GRU je state a output to samé, ale je to tam odlišeně kvůli LSTM, která to má zvlášť (typicky značené jako _h_ a _c_).

Reply:
Ano, prev_rnn_state se v GRU dekoderu nepouziva. Nicméně to je potřeba feedovat když děláš beam search.. #550 by tohle mělo vyřešit

Reply:
Ok, tak tady jde spis o terminologii. Preferoval bych predavat do GRU prev_rnn_state, vzhledem k tomu, ze i definice __call__(inputs, state, scope=None) to nazyva "state" (je to imho mene matouci).

Reply:
Problém je hlavně v tom, že se slovo state používá na hrozně moc místech. Po #550 snad nebude tohle rozlišení vůbec potřeba.